### PR TITLE
Ncgames 4 get api reviews

### PR DESCRIPTION
**Pull request by coolpat1993**
added error handling for review id, checking that the input is valid and a number. added similar tests

### CODE REVIEW

File: __tests__/api.test.js [hunk 1]
```diff
@@ -43,3 +43,38 @@ describe("GET api/categories", () => {
       });
   });
 });
+
+describe.only("2. GET /api/reviews/:review_id", () => {
```
Comment (emilyb93) [inline __tests__/api.test.js]: no .onlys

File: __tests__/api.test.js [hunk 2]
```diff
@@ -43,3 +43,38 @@ describe("GET api/categories", () => {
...
+      .get(`/api/reviews/${review_id}`)
+      .expect(400)
+      .then(response => {
+        expect(response.body).toEqual({
+          status: 400,
+          msg: "invalid review ID",
+        });
+      });
+  });
+});
```
Comment (emilyb93) [inline __tests__/api.test.js:92]: Id like to see a test for if the id is a number but it doesnt yet exist in the database

File: models/models.js [hunk 3]
```diff
@@ -5,3 +5,15 @@ exports.selectCategories = () => {
     return result.rows;
   });
 };
+
+exports.selectReviews = review_id => {
+  if (isNaN(review_id) === true) {
+    return Promise.reject({ status: 400, msg: "invalid review ID" });
```
Comment (emilyb93) [inline models/models.js:12]: SQL will throw its own error for this situation from db.query (if the data type is not correct for the column), i would lean more towards letting SQL deal with this and catching the error and then acting accordingly

**Review by emilyb93 — CHANGES_REQUESTED** [review]
Another test needed and please remove .only

File: models/models.js [hunk 4]
```diff
@@ -1,7 +1,34 @@
...
+      .then(reviewLength => {
+        if (review_id > 0 && review_id <= reviewLength) {
+          return db
+            .query("SELECT * FROM reviews WHERE review_id = $1", [review_id])
+            .then(result => {
+              return result.rows[0];
+            });
+        } else {
+          return Promise.reject({
+            status: 200,
```
Comment (hevdevs) [inline models/models.js]: should be expecting status 404 not found here

File: models/models.js (hunk 3 above)
Comment (hevdevs) [reply models/models.js:12]: this condition check should be unnecessary as Emily mentioned, do we already have a catch block in our controller and error handling for a 400/bad path in our app.js?

**Review by hevdevs — CHANGES_REQUESTED** [review]
just a couple of little bits to adjust here!

**Review by emilyb93 — APPROVED** [review]
Seems good to go :)

**Pull request merged**